code/game/objects/structures/tables_racks.dm

```diff
@@ -305,9 +305,8 @@
 			to_chat(user, SPAN_WARNING("You slice at the table, but only claw it up a little."))
 		return
 
-	if (istype(W, /obj/item/device/sentry_computer))
-		var/obj/item/device/sentry_computer/computer = W
-		computer.setup(src)
+	if (W.has_special_table_placement)
+		W.set_to_table(src)
```

Comment:
do this check before any istypes etc to save on proc overhead, easier/more generic checks should come first in big procs

Reply:
done, moved to start, just below if(!W)